tikv-client/src/main/java/com/pingcap/tikv/predicates/TiKVScanAnalyzer.java

```diff
@@ -297,47 +297,116 @@ TiKVScanPlan buildIndexScan(
     }
   }
 
-  private Pair<Key, Key> buildTableScanKeyRangePerId(long id, IndexRange ir) {
-    Key startKey;
-    Key endKey;
+  private ScanRange buildTableScanKeyRangePerId(long id, IndexRange ir) {
+    boolean isUnsignedLong = false;
+    ScanRange range = new ScanRange();
+    Key signedStartKey = null, signedEndKey = null, unsignedStartKey = null, unsignedEndKey = null;
     if (ir.hasAccessKey()) {
       checkArgument(
           !ir.hasRange(), "Table scan must have one and only one access condition / point");
-
       Key key = ir.getAccessKey();
       checkArgument(key instanceof TypedKey, "Table scan key range must be typed key");
-      TypedKey typedKey = (TypedKey) key;
-      startKey = RowKey.toRowKey(id, typedKey);
-      endKey = startKey.next();
+      /**
+       * {@link com.pingcap.tikv.types.IntegerType#decodeNotNull(int, CodecDataInput)}
+       */
+      Object obj = ((TypedKey) key).getValue();
+      if (obj instanceof Long) {
+        signedStartKey = RowKey.toRowKey(id, new IntHandle((long) obj));
+        signedEndKey = signedStartKey.next();
+      } else if (obj instanceof BigDecimal) {
+        BigDecimal bigDecimal = (BigDecimal) obj;
+        signedStartKey = RowKey.toRowKey(id, new IntHandle(bigDecimal.longValue()));
+        signedEndKey = signedStartKey.nextPrefix();
+      }
     } else if (ir.hasRange()) {
       checkArgument(
           !ir.hasAccessKey(), "Table scan must have one and only one access condition / point");
       Range<TypedKey> r = ir.getRange();
-
+      BigDecimal max = new BigDecimal(Long.MAX_VALUE);
       if (!r.hasLowerBound()) {
         // -INF
-        startKey = RowKey.createMin(id);
+        signedStartKey = RowKey.createMin(id);
```

Comment:
signedStartKey need to be 0 when unsigned long. 
for example : id < 2 => [0,2)

Reply:
changed,when no upper bound,signedStartKey will be zero